keras/utils/metrics_utils.py

```diff
@@ -174,7 +174,13 @@ def merge_fn_wrapper(distribution, merge_fn, *args):
                 # Wrapping result in identity so that control dependency between
                 # update_op from `update_state` and result works in case result
                 # returns a tensor.
-                return tf.identity(result)
+                if isinstance(result, dict):
+                    return {
+                        key: array_ops.identity(value)
```

Comment:
Note that keras use the public TF API. This should be tf.identity(), and array_ops is a private TF API.